### PR TITLE
Patched build.rs to run on systems without rustup. Added flake for ni…

### PR DESCRIPTION
**Pull request by benwis**
In order to build spin on systems without rustup, you can't use rustup to check for targets. I've replaced that check with one from rustc. If it's not installed it still fails for nix users as it can't install it, but if it is it passes at least

### CODE REVIEW

**Comment by itowlson** [comment]
@benwis Could you sign off the commit (and GPG sign if not already done) please?  Thanks!

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+{
+  description = "A basic Rust devshell";
```
Comment (endocrimes) [inline flake.nix:3]: We should probably update this to reference spin, but that can be done as a follow up

**Review by endocrimes — COMMENTED** [review]
mostly lgtm, I'm ooo rn but I can give this a try tonight

File: flake.nix (hunk 1 above)
Comment (benwis) [reply flake.nix]: I mentioned spin in the new commit

**Comment by benwis** [comment]
> @benwis Could you sign off the commit (and GPG sign if not already done) please? Thanks!

I will try, I haven't setup code signing before, shouldn't be too bad

**Comment by benwis** [comment]
> @benwis Could you sign off the commit (and GPG sign if not already done) please? Thanks!

Signed!

**Review by endocrimes — APPROVED** [review]
Tested that this works correctly on linux - Darwin test is taking a longer because well... a laptop is not a server 😅 - I'll follow up with anything needed for nix-darwin builds though.

**Comment by benwis** [comment]
> Tested that this works correctly on linux - Darwin test is taking a longer because well... a laptop is not a server 😅 - I'll follow up with anything needed for nix-darwin builds though.

Thank you!

**Pull request merged**